src/servers/Server_Zone/Script/NativeScript.cpp

```diff
@@ -0,0 +1,139 @@
+#include "NativeScript.h"
+
+namespace Core {
+    namespace Scripting {
+
+        NativeScript::NativeScript( )
+        { }
+
+        StatusEffectScript* NativeScript::getStatusEffectScript( uint32_t statusId )
+        {
+           auto script = m_statusEffectScripts.find( statusId );
+           if( script == m_statusEffectScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        ActionScript* NativeScript::getActionScript(uint32_t abilityId)
+        {
+           auto script = m_actionScripts.find( abilityId );
+           if( script == m_actionScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        QuestScript* NativeScript::getQuestScript( uint32_t questId )
+        {
+           auto script = m_questScripts.find( questId );
+           if( script == m_questScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        BattleNpcScript* NativeScript::getBattleNpcScript( uint32_t npcId )
+        {
+           auto script = m_battleNpcScripts.find( npcId );
+           if( script == m_battleNpcScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        ZoneScript* NativeScript::getZoneScript( uint32_t zoneId )
+        {
+           auto script = m_zoneScripts.find( zoneId );
+           if( script == m_zoneScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        bool NativeScript::loadScript( std::string path )
+        {
+           auto info = m_loader.loadModule( path );
+           if( info )
+           {
+              // todo: this is shit
+              if( auto script = m_loader.getScriptObject< StatusEffectScript >( info->handle, "StatusEffectScript" ) )
```

Comment:
you're aware of this possible improvement but you're better off making a helper method "hasValidExports", passing the latter and having a switch case. 
Also "StatusEffectScript", "ActionScript" etc -> enum it. so you can just use hasValidExports( ScriptType scriptType )